modules/miner/poolmanager.go

```diff
@@ -0,0 +1,292 @@
+	cryptoSig, err := crypto.SignHash(sigHash, sk)
+	if err != nil {
+		return err
+	}
+	refundTxn.TransactionSignatures[1].Signature = cryptoSig[:]
+
+	// Send the refundTxn back to the pool
+	err = encoding.WriteObject(conn, refundTxn)
+	if err != nil {
+		return err
+	}
+
+	// TODO: Verify the channelTxn has been signed and broadcasted
+
+	// Send the pool an address so it can pay us
+	err = encoding.WriteObject(conn, m.address)
+	if err != nil {
+		return err
+	}
+
+	m.poolSK = sk
+	fmt.Println("Miner completed the payment channel")
+	return nil
+}
+
+// Closes the specified payment channel by broadcasting the final transaction
+// to the network. The miner recieves its payouts, but prevents more money from
+// being sent through this channel
+func (m *Miner) closeChannel() error {
+	// TODO: First, make sure there is a channel to close?
+	// Sign and broadcast the channel's output transaction
+	sigHash := m.poolTransaction.SigHash(1)
+	cryptoSig, err := crypto.SignHash(sigHash, m.poolSK)
+	if err != nil {
+		return err
+	}
+	m.poolTransaction.TransactionSignatures[1].Signature = cryptoSig[:]
+	err = m.tpool.AcceptTransactionSet([]types.Transaction{m.poolTransaction})
+	if err != nil {
+		return err
+	}
+
+	// TODO: Tell pool that we closed the channel
+	return nil
+}
+
+// Reconstruct a block using its pool header by getting the block from the
+// block manager, then changing the payouts.
+func (m *Miner) reconstructPoolBlock(ph types.BlockHeader) (types.Block, error) {
+	var zeroNonce [8]byte
+	poolHeader := ph
+	poolHeader.Nonce = zeroNonce
+
+	// Convert poolHeader to a header that the block manager knows about
+	lookupBH, exists := m.poolHeaderMem[poolHeader]
+	if !exists {
+		return types.Block{}, errors.New("Header is not a valid poolheader or is too old")
+	}
+
+	block, err := m.reconstructBlock(lookupBH)
+	if err != nil {
+		return types.Block{}, err
+	}
+
+	subsidy := block.MinerPayouts[0].Value
+	minerPayout := subsidy.Div(types.NewCurrency64(100)).Mul(types.NewCurrency64(uint64(m.minerPercentCut)))
+	poolPayout := subsidy.Sub(minerPayout)
+	blockPayouts := []types.SiacoinOutput{
+		types.SiacoinOutput{Value: minerPayout, UnlockHash: m.address},
+		types.SiacoinOutput{Value: poolPayout, UnlockHash: m.poolPayoutAddress}}
+
+	newBlock := types.Block{
+		ParentID:     block.ParentID,
+		Timestamp:    block.Timestamp,
+		MinerPayouts: blockPayouts,
+		Transactions: block.Transactions,
+	}
+
+	return newBlock, nil
+}
+
+// Connects to the pool hosted at the given ip. The miner negotiates a payment
+// channel and gets certain values from the pool, like the payout address(es)
+// and payout ratios (what percent goes to who)
+func (m *Miner) ConnectToPool(ip string) error {
+	fmt.Println("connect to pool: ", ip)
+	conn, err := net.DialTimeout("tcp", ip, 10e9)
+	if err != nil {
+		return err
+	}
+	defer conn.Close()
+	m.poolIP = ip
+
+	err = encoding.WriteObject(conn, [8]byte{'S', 'e', 't', 't', 'i', 'n', 'g', 's'})
```

Comment:
should probably be defined as a variable in modules/miner.go.
I recommend defining it as a `types.Specifier`.